fail2ban/server/filter.py

```diff
@@ -518,19 +526,45 @@ def findFailure(self, tupleLine, date=None, returnRawHost=False,
 						 % ("\n".join(failRegex.getMatchedLines()), timeText))
 				else:
 					self.__lineBuffer = failRegex.getUnmatchedTupleLines()
+					# retrieve failure-id, host, etc from failure match:
+					raw = returnRawHost
 					try:
-						host = failRegex.getHost()
-						if returnRawHost:
-							failList.append([failRegexIndex, IPAddr(host), date,
-								 failRegex.getMatchedLines()])
+						fail = failRegex.getGroups()
+						# failure-id:
+						fid = fail.get('fid')
+						# ip-address or host:
+						host = fail.get('ip4')
```

Comment:
Exactly: now, the pure IPs would be converted to IPAddr directly and only hosts (dns), if not raw, will be expanded to list of IPs via textToIp or ignored corresonding value of usedns;

BTW: I don't like such sugar, that "breaks" thoughts logic, as if it were boolean...